### DISCUSSION TITLE
Misunderstanding about inheritance

### DISCUSSION THREAD

**Discussion by Sutterlet-Bruno** [capnproto/capnproto#1870] [Q&A]
I am currently working on a project that uses cap'n proto framework. 

I am trying to use inheritance to pass methods and structs from and interface to others, 

here is an example : 
Schema : 
 ```bash
@0xf5e7f62c0dfaa9d1;

interface A {


}

interface B extends(A){
    foo @0 (bar :Text);
}
```

Compiled structs in c++ :
 ```cpp
struct A {
  A() = delete;

#if !CAPNP_LITE
  class Client;
  class Server;
#endif  // !CAPNP_LITE

  struct FooParams;
  struct FooResults;

  #if !CAPNP_LITE
  struct _capnpPrivate {
    CAPNP_DECLARE_INTERFACE_HEADER(a5b79a44e8a06f04)
    static constexpr ::capnp::_::RawBrandedSchema const* brand() { return &schema->defaultBrand; }
  };
  #endif  // !CAPNP_LITE
};

[...]

struct B {
  B() = delete;

#if !CAPNP_LITE
  class Client;
  class Server;
#endif  // !CAPNP_LITE

  /* no trace of FOO */

  #if !CAPNP_LITE
  struct _capnpPrivate {
    CAPNP_DECLARE_INTERFACE_HEADER(a3fd1807709cfb96)
    static constexpr ::capnp::_::RawBrandedSchema const* brand() { return &schema->defaultBrand; }
  };
  #endif  // !CAPNP_LITE
};
```
Here B didn't inherit the method FOO, I thought it would work the same way as in C++ but I think it's not.

What does exactly  `extends` ?

**Comment by kentonv** [comment]
Take a look at the `Client` and `Server` types. You'll find that `B::Client` inherits `A::Client`, and `B::Server` inherits `A::Server`.

The structs `B` and `A` are really just used as namespaces, to contain `Client` and `Server`. `B` and `A` cannot be instantiated directly.

**Reply by Sutterlet-Bruno** [reply]
Thanks for replying, I completely missed this while searching.
